streampark-console/streampark-console-service/src/test/java/org/apache/streampark/base/SpringTestBase.java

```diff
@@ -0,0 +1,126 @@
+ * limitations under the License.
+ */
+
+package org.apache.streampark.base;
+
+import org.apache.streampark.common.conf.CommonConfig;
+import org.apache.streampark.common.conf.ConfigConst;
+import org.apache.streampark.console.StreamParkConsoleBootstrap;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureTestEntityManager;
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
+import org.testcontainers.containers.BrowserWebDriverContainer;
+import org.testcontainers.containers.MockServerContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+
+/** base tester. */
+@TestPropertySource(locations = {"classpath:application-test.yml"})
+@ActiveProfiles("test")
+@ExtendWith({MockitoExtension.class, SpringExtension.class})
+@Transactional
+@SpringBootTest(
+    classes = StreamParkConsoleBootstrap.class,
+    webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@Testcontainers
+@AutoConfigureWebTestClient(timeout = "60000")
+@AutoConfigureTestEntityManager
+public abstract class SpringTestBase {
+
+  protected static final Logger LOG = LoggerFactory.getLogger(SpringTestBase.class);
+
+  protected static final DockerImageName MOCKSERVER_IMAGE =
+      DockerImageName.parse("mockserver/mockserver").withTag("mockserver-5.5.4");
+
+  // Use it to test on chrome.
+  @Container
+  public static BrowserWebDriverContainer<?> chrome =
+      new BrowserWebDriverContainer<>()
+          .withCapabilities(new ChromeOptions())
+          .withLogConsumer(new Slf4jLogConsumer(LOG));
+
+  // Use it to mock external services.
+  @Container
+  public static MockServerContainer mockServer =
+      new MockServerContainer(MOCKSERVER_IMAGE).withLogConsumer(new Slf4jLogConsumer(LOG));
```

Comment:
In fact, I prefer to move these two suites down to an abstract subclass. If a test class does not use these two suites, but still inherits the current class, it is likely to slow down the startup speed of the test class.

Reply:
<img width="1690" alt="image" src="https://user-images.githubusercontent.com/64569824/222904473-89aa7fe5-5f5d-4465-8694-0c36bdff4cf0.png">